### PR TITLE
Add validation and uploading of jobs configuration

### PR DESCRIPTION
**Pull request by michal-hudy**
**Description**

Changes proposed in this pull request:

- Added a simple application that uploads configuration to Prow
- Enabled validation of job configs

**Related issue(s)**

See #87

### CODE REVIEW

File: development/tools/cmd/configuploader/main.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	config, err := configMapFromFile(name, path)
+	if err != nil {
+		return err
+	}
+
+	_, err = client.Update(config)
+	return err
+}
+
+func uploadFromFiles(name, path string, client configMapSetter) error {
```
Comment (aszecowka) [inline development/tools/cmd/configuploader/main.go]: consider renaming: updateConfigMapFromDirecotry and uploadFromFile --> updateConfigMapFromFile.
Additionally, it is updating or replacing?

File: development/tools/cmd/configuploader/main_test.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+const (
+	dirStructure = "where/is/my/test"
+)
+
+var files = [...]string{"where/fileA.yaml", "where/is/fileB.yaml", "where/is/my/fileC.yaml"}
+
+func TestUploadFromFile(t *testing.T) {
+	//given
+	tmpDir, err := ioutil.TempDir("", "TestFindAllRec")
```
Comment (aszecowka) [inline development/tools/cmd/configuploader/main_test.go]: minor: prefix should be the same as a test name

File: development/tools/pkg/file/file.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package file
+
+import (
+	"os"
+	"path/filepath"
+)
+
+func FindAllRec(rootPath, extension string) ([]string, error) {
```
Comment (aszecowka) [inline development/tools/pkg/file/file.go]: name of this function is cryptic for me

File: development/tools/cmd/configuploader/main.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+		},
+		Data: map[string]string{
+			filename: string(data),
+		},
+	}
+
+	return &result, nil
+}
+
+func configMapFromFiles(name, rootPath string) (*v1.ConfigMap, error) {
```
Comment (aszecowka) [inline development/tools/cmd/configuploader/main.go]: minor: maybe better name will be: configMapFromYamlsInDir

File: development/tools/cmd/configuploader/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# Config Uploader
+
+## Overview
+
+This command is responsible for uploading Prow configuration.
+
+## Usage
+
+To run, use:
```
Comment (aszecowka) [inline development/tools/cmd/configuploader/README.md]: Also, add information that we make an assumption that configmaps already exist

File: development/tools/cmd/configuploader/README.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Config Uploader
+
+## Overview
+
+This command is responsible for uploading Prow configuration.
+
+## Usage
```
Comment (aszecowka) [inline development/tools/cmd/configuploader/README.md]: If this tool has to be used for production cluster, please update documentation how to configure production cluster

File: prow/jobs/test-infra/validation.jobs.yaml [hunk 7]
```diff
@@ -1,23 +1,24 @@
...
+    - name: prow/test-infra/validate-scripts
+      run_if_changed: "/*.sh$"
+      decorate: true
+      skip_report: false
+      spec:
+        containers:
+          - image: eu.gcr.io/kyma-project/prow/test-infra/develop/bootstrap:v20181115-af8fb73
+            command:
+              - "/home/prow/go/src/github.com/kyma-project/test-infra/development/validate-scripts.sh"
+    - name: prow/test-infra/validate-configs
```
Comment (aszecowka) [inline prow/jobs/test-infra/validation.jobs.yaml]: we should have a job to check if config uploader compiles - please fix it here or create a follow up task

File: development/tools/cmd/configuploader/README.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
+# Config Uploader
+
+## Overview
+
+This command is responsible for uploading Prow configuration.
```
Comment (kazydek) [inline development/tools/cmd/configuploader/README.md]:
```suggestion
This command is responsible for uploading Prow configuration to a Prow cluster.
```

File: development/tools/cmd/configuploader/README.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
```
[the remark this answers is not in the document]
Comment (michal-hudy) [reply development/tools/cmd/configuploader/README.md]: Added more information.

File: development/tools/cmd/configuploader/README.md (hunk 6 above)
Comment (michal-hudy) [reply development/tools/cmd/configuploader/README.md]: Updated

File: development/tools/cmd/configuploader/main.go [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+	config, err := configMapFromFiles(name, path)
+	if err != nil {
+		return err
+	}
+
+	_, err = client.Update(config)
+	return err
+}
+
+func configMapFromFile(name, path string) (*v1.ConfigMap, error) {
```
[the remark this answers is not in the document]
Comment (michal-hudy) [reply development/tools/cmd/configuploader/main.go]: Renamed

File: prow/jobs/test-infra/validation.jobs.yaml (hunk 7 above)
Comment (michal-hudy) [reply prow/jobs/test-infra/validation.jobs.yaml]: Follow up created: #134

File: development/tools/cmd/configuploader/README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (aszecowka) [reply development/tools/cmd/configuploader/README.md]: My proposal: Use it for a newly created Prow cluster or to update changes in the configuration on a cluster from a forked repository that does not use config-updater?

**Pull request merged**